crates/cxx-qt-lib/src/core/qtypes.rs

```diff
@@ -0,0 +1,253 @@
+// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use cxx::{type_id, ExternType};
+use std::ops::{Deref, DerefMut};
+
+#[cxx::bridge]
+mod ffi {
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/qtypes.h");
+    }
+}
+
+/// Typedef for long long int. This type is guaranteed to be 64-bit on all platforms supported by Qt.
+#[repr(transparent)]
+#[derive(Default, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct QInt64(i64);
+
+impl Deref for QInt64 {
+    type Target = i64;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for QInt64 {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+impl From<i64> for QInt64 {
+    fn from(value: i64) -> Self {
+        Self(value)
+    }
+}
+
+impl From<QInt64> for i64 {
+    fn from(value: QInt64) -> Self {
+        value.0
+    }
+}
+
+// Safety:
+//
+// Static checks on the C++ side to ensure the size is the same.
+unsafe impl ExternType for QInt64 {
+    type Id = type_id!("qint64");
+    type Kind = cxx::kind::Trivial;
+}
+
+/// Integral type for representing pointers in a signed integer (useful for hashing, etc.).
+#[repr(transparent)]
+#[derive(Default, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
```

Comment:
Again, let's add `Copy` and `Clone` to that list :)

Reply:
done